In [1]:
%load_ext autoreload
%autoreload 2

from game import Game
from contest import Contest

In [2]:
# game = Game(n_players=4, n_mafia=1, model_mafia="google/gemini-flash-1.5", model_townsperson="google/gemini-2.0-flash-lite-001", player_names=["Jill", "John", "Jane", "Jim"])

In [3]:
# await game.run()

In [2]:
names = [
    "Alice",
    "Bob",
    "Charlie",
    "Diana",
    "Eve",
    "Frank",
    "Tom",
    "Mary",
    "Steve",
    "Ava",
    "Pearl",
    "Jill",
    "John",
    "George",
    "Lily",
    "Olivia",
    "Jane"
  ]

In [3]:
#go with 8 players, 2 mafia

contest = Contest("test_few", n_games=2,n_concurrent_games=50, n_players=4, n_mafia=1, model_a="google/gemini-flash-1.5", model_b="google/gemini-flash-1.5-8b", player_names=names)

In [4]:
resuts = await contest.run()

2025-03-25 17:39:01,448 - game_0 - INFO - Running phase: Phase.INTRO on day 0
2025-03-25 17:39:01,453 - game_1 - INFO - Running phase: Phase.INTRO on day 0


{'Alice': Alice (role: townsperson, model=meta-llama/llama-3.2-1b-instruct), 'Bob': Bob (role: mafia, model=google/gemini-flash-1.5), 'Charlie': Charlie (role: townsperson, model=meta-llama/llama-3.2-1b-instruct), 'Diana': Diana (role: townsperson, model=meta-llama/llama-3.2-1b-instruct)}
{'Alice': Alice (role: townsperson, model=google/gemini-flash-1.5), 'Bob': Bob (role: townsperson, model=google/gemini-flash-1.5), 'Charlie': Charlie (role: townsperson, model=google/gemini-flash-1.5), 'Diana': Diana (role: mafia, model=meta-llama/llama-3.2-1b-instruct)}


Rate Limiting Stats (last 10.0s):
  Requests per second: 0.6
  Current RPS: 60.0
2025-03-25 17:39:15,732 - game_1 - INFO - Running phase: Phase.DAY on day 0
2025-03-25 17:39:18,847 - game_0 - INFO - Running phase: Phase.DAY on day 0
Rate Limiting Stats (last 10.0s):
  Requests per second: 1.0
  Current RPS: 60.0
2025-03-25 17:39:32,055 - game_1 - INFO - Diana made an invalid vote: Charlie.
2025-03-25 17:39:32,056 - game_1 - INFO - Eliminated player: Bob


The mafia have voted on who to eliminate as follows: {'Alice': 'Bob', 'Bob': 'Alice', 'Charlie': 'Bob'}. The player who was eliminated was Bob.


Rate Limiting Stats (last 10.0s):
  Requests per second: 1.7
  Current RPS: 60.0
2025-03-25 17:39:35,986 - game_1 - INFO - Running phase: Phase.NIGHT on day 0
2025-03-25 17:39:36,666 - game_0 - INFO - Alice made an invalid vote: Bob.
2025-03-25 17:39:36,667 - game_0 - INFO - Charlie made an invalid vote: **Bob**
2025-03-25 17:39:36,667 - game_0 - INFO - Diana made an invalid vote: Alice.
2025-03-25 17:39:36,668 - game_0 - INFO - Eliminated player: Alice


The mafia have voted on who to eliminate as follows: {'Bob': 'Alice'}. The player who was eliminated was Alice.


2025-03-25 17:39:40,577 - game_1 - INFO - Diana made an invalid vote: with?
2025-03-25 17:39:40,578 - game_1 - WARNING - No valid votes cast
2025-03-25 17:39:40,578 - game_1 - INFO - Running phase: Phase.DAY on day 1
2025-03-25 17:39:40,818 - game_0 - INFO - Running phase: Phase.NIGHT on day 0
Rate Limiting Stats (last 10.0s):
  Requests per second: 1.1
  Current RPS: 60.0
2025-03-25 17:39:47,542 - game_0 - INFO - Eliminated player: Charlie
2025-03-25 17:39:50,271 - game_0 - INFO - Game over!


Game over. Results: {'model_mafia': 'google/gemini-flash-1.5', 'model_townsperson': 'meta-llama/llama-3.2-1b-instruct', 'winner': 'mafia', 'mafia_invalid_votes': 0, 'townsperson_invalid_votes': 3, 'mafia_total_time': 29.171844244003296, 'townsperson_total_time': 45.64077568054199, 'mafia_total_messages': 10, 'townsperson_total_messages': 18, 'n_players': 4, 'n_mafia': 1, 'game_duration': 48.82504791600513, 'game_rounds': 1}


2025-03-25 17:39:53,723 - game_1 - INFO - Diana made an invalid vote: with?
2025-03-25 17:39:53,724 - game_1 - INFO - Eliminated player: Diana


The mafia have voted on who to eliminate as follows: {'Alice': 'Diana', 'Charlie': 'Diana'}. The player who was eliminated was Diana.
Game over. Results: {'model_mafia': 'meta-llama/llama-3.2-1b-instruct', 'model_townsperson': 'google/gemini-flash-1.5', 'winner': 'townsperson', 'mafia_invalid_votes': 3, 'townsperson_invalid_votes': 0, 'mafia_total_time': 20.916424036026, 'townsperson_total_time': 74.18651413917542, 'mafia_total_messages': 12, 'townsperson_total_messages': 25, 'n_players': 4, 'n_mafia': 1, 'game_duration': 55.82733254201594, 'game_rounds': 1}


In [5]:
resuts

ContestStats(name='test_few', n_games=4, n_players=8, n_mafia=2, contest_duration=212.5934510419902, avg_game_duration=161.68392218750523, game_parallelism=50, time_finished=1742923859.3759968, model_a_stats=ModelContestStats(model_name='google/gemini-flash-1.5', n_wins_mafia=2, n_wins_townsperson=1, n_invalid_votes=0, total_time=869.2859044075012, total_messages=273), model_b_stats=ModelContestStats(model_name='meta-llama/llama-3.2-1b-instruct', n_wins_mafia=1, n_wins_townsperson=0, n_invalid_votes=42, total_time=512.585022687912, total_messages=196))

In [6]:
resuts.summary()

{'model_win_rates': {'google/gemini-flash-1.5': 0.75,
  'meta-llama/llama-3.2-1b-instruct': 0.25},
 'role_win_rates': {'mafia': 0.75, 'townsperson': 0.25},
 'model_invalid_votes': {'google/gemini-flash-1.5': 0,
  'meta-llama/llama-3.2-1b-instruct': 42},
 'model_latency': {'google/gemini-flash-1.5': 3.18,
  'meta-llama/llama-3.2-1b-instruct': 2.62},
 'contest_duration': 212.5934510419902,
 'avg_game_duration': 161.68392218750523}

# Results
- 2 mafia, 10 players: 54% town. 
- 2 mafia, 9 players: 60% town. with 150 questions, 63% town
- 2 mafia, 8 players: 38% town. with 150 questions; 42% town. 